core/src/main/java/org/apache/carbondata/core/preagg/TimeSeriesUDF.java

```diff
@@ -119,10 +150,27 @@ private void initialize() {
     if (TIMESERIES_FUNCTION.isEmpty()) {
       TIMESERIES_FUNCTION.add("second");
       TIMESERIES_FUNCTION.add("minute");
+      TIMESERIES_FUNCTION.add("five_minute");
+      TIMESERIES_FUNCTION.add("ten_minute");
+      TIMESERIES_FUNCTION.add("fifteen_minute");
+      TIMESERIES_FUNCTION.add("thirty_minute");
       TIMESERIES_FUNCTION.add("hour");
       TIMESERIES_FUNCTION.add("day");
+      TIMESERIES_FUNCTION.add("week");
       TIMESERIES_FUNCTION.add("month");
       TIMESERIES_FUNCTION.add("year");
     }
+    int firstDayOfWeek;
+    try {
+      firstDayOfWeek = DaysOfWeekEnum.valueOf(CarbonProperties.getInstance()
+          .getProperty(CarbonCommonConstants.CARBON_TIMESERIES_FIRST_DAY_OF_WEEK,
+              CarbonCommonConstants.CARBON_TIMESERIES_FIRST_DAY_OF_WEEK_DEFAULT).toUpperCase())
+          .getOrdinal();
+    } catch (IllegalArgumentException ex) {
+      firstDayOfWeek = DaysOfWeekEnum.valueOf(CarbonProperties.getInstance()
```

Comment:
Need to add log that user set wrong value in `CarbonCommonConstants.CARBON_TIMESERIES_FIRST_DAY_OF_WEEK` ?